src/Microsoft.ML.Core/Data/ColumnType.cs

```diff
@@ -640,23 +644,14 @@ public static bool IsValidDataType(Type type)
         }
 
         /// <summary>
-        /// This is the Min of the key type for display purposes and conversion to/from text. The values
-        /// actually stored always start at 1 (for the smallest legal value), with zero being reserved
-        /// for "not there"/"none". Typical Min values are 0 or 1, but can be any value >= 0.
+        /// If this key type has a known cardinality, Count is that cardinality. Otherwise, this returns
```

Comment:
>Otherwise [](start = 81, length = 9)

I am wondering if there is an advantage to make the distinction between unknown cardinality and Count=ulong.MaxValue. In the Parse methods it seems like there doesn't need to be a distinction. Are there scenarios where the KeyType would behave differently for Count=0 vs. Count=ulong.MaxValue?

Reply:
One of the things we see in the original issue is, we want to get rid of the concept of the "unknown cardinality" altogether.

Reply:
As you and Tom pointed out, I am going to remove the option of an unknown cardinality for KeyType. I will replace that scenario with the maximum available value for count given the datatype used.

---
In reply to: [248106490](https://github.com/dotnet/machinelearning/pull/2146#discussion_r248106490) [](ancestors = 248106490)

Reply:
Ok that makes sense I will remove that option!

---
In reply to: [248367868](https://github.com/dotnet/machinelearning/pull/2146#discussion_r248367868) [](ancestors = 248367868)